panel/src/panel/dialog.js

```diff
@@ -88,6 +89,13 @@ export default (panel) => {
 				return this.openComponent(dialog);
 			}
 
+			// handle drawer object with url property
+			if (isObject(dialog) && dialog.url) {
+				options = dialog;
+				dialog = dialog.url;
+				delete options.url;
+			}
```

Comment:
`Explanation:` This sadly isn't fully DRY but repeated for the drawer as well. However, we can not simply move it to `modal.open()` as then the URL prefixing code below, which is different for dialogs and drawers, would not be taken into account.